### PR TITLE
Fix small issues in stats_exporter integration

### LINKED ISSUE #1002: The `statsd_exporter` integration don't map timers (observers)
I am using the Grafana Agent version 0.19.0 with the `statsd_exporter` integration `enabled`, which the timer (observer) metrics have been ignored. With debug enabled, it is possible to see that all other types are mapped/processed. The same behavior can be seen at [http://localhost:9090/integrations/statsd_exporter/metrics](http://localhost:9090/integrations/statsd_exporter/metrics).

The metrics are sent via UTP, and when using the Prometheus standalone StatsD exporter, the timers (observers) are automatically mapped to summaries.

For test purposes, the version used of the Prometheus standalone StatsD exporter was 0.22.2, and no extra parameters were used.  For the `statsd_exporter` integration only with:
```
statsd_exporter:
    enabled: true
    listen_udp: :9125
```

### LINKED ISSUE #1052: `mapping_config` causes `invalid memory address or nil pointer dereference`
After update the agent from 0.19.0 to 0.20.0, when `statsd_exporter` is enable with `mappings`, the agent fail to start with the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x2d8a9fb]

goroutine 1 [running]:
github.com/go-kit/log.(*context).Log(0xc0003d0370, 0xc00033d480, 0x2, 0x2, 0x5, 0x0)
        /go/pkg/mod/github.com/go-kit/log@v0.2.0/log.go:168 +0x2db
github.com/prometheus/statsd_exporter/pkg/mapper.(*MetricMapper).InitFromYAMLString(0xc0007d0780, 0xc001280000, 0x214, 0x0, 0x0)
        /go/pkg/mod/github.com/prometheus/statsd_exporter@v0.22.2/pkg/mapper/mapper.go:169 +0x10f0
github.com/grafana/agent/pkg/integrations/statsd_exporter.New(0x468e1e0, 0xc000f5f090, 0xc000050d20, 0x0, 0x0, 0x0, 0x0)
        /drone/src/pkg/integrations/statsd_exporter/statsd_exporter.go:131 +0x9d4
github.com/grafana/agent/pkg/integrations/statsd_exporter.(*Config).NewIntegration(0xc000050d20, 0x468e1e0, 0xc000f5f090, 0x2, 0x2, 0x468e1e0, 0xc000f5f090)
        /drone/src/pkg/integrations/statsd_exporter/statsd_exporter.go:94 +0x3f
github.com/grafana/agent/pkg/integrations.(*Manager).ApplyConfig(0xc0011fa4e0, 0x10101, 0xc0008fa540, 0x3, 0x4, 0x0, 0xc0006967f8, 0x1, 0x1, 0x12a05f200, ...)
        /drone/src/pkg/integrations/manager.go:239 +0x129d
github.com/grafana/agent/pkg/integrations.NewManager(0x10101, 0xc0008fa540, 0x3, 0x4, 0x0, 0xc0006967f8, 0x1, 0x1, 0x12a05f200, 0x3039, ...)
        /drone/src/pkg/integrations/manager.go:186 +0x1e5
main.NewEntrypoint(0xc000611d10, 0xc00077a000, 0xc0009bfd80, 0xc000c0ca10, 0x4419e5, 0xc000084710)
        /drone/src/cmd/agent/entrypoint.go:93 +0x51e
main.main()
        /drone/src/cmd/agent/main.go:66 +0x1dd
```
The agent integration configuration example:
```
integrations:
  statsd_exporter:
    enabled: true
    mapping_config:
      mappings:
        - match: jvm_buffer_count
          match_type: regex
          help: An estimate of the number of buffers in the pool
          name: jvm_buffer_count_buffers
          match_metric_type: gauge
```
P.S.: The same setting above was working on 0.19.0 with the InfluxDB tags.

### PR DESCRIPTION
**Pull request by gaantunes**
#### PR Description 
* Adding logger instance to Statsd integration mapper instantiation.
* Update statsd dependency to include fix where custom registry was not used 
* Don't create a cache when the cache size is 0 
#### Which issue(s) this PR fixes 
Fixes #1052 
Fixes #1002 
#### PR Checklist

- [X] CHANGELOG updated 
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Main (unreleased)
 
+- [BUGFIX] Statsd Integration: Adding logger instance to the statsd mapper instantiation. (@gaantunes)
```
Comment (rfratto) [inline CHANGELOG.md:3]: Hey, can you retain the grouping of FEATURE -> ENHANCEMENT -> BUGFIX -> CHANGE -> DEPRECATION? And then newer changeling entries get added to the bottom of the group, so they're sorted by group & time.

File: pkg/integrations/statsd_exporter/statsd_exporter.go [hunk 2]
```diff
@@ -125,7 +125,10 @@ func New(log log.Logger, c *Config) (integrations.Integration, error) {
 	if c.ListenUDP == "" && c.ListenTCP == "" && c.ListenUnixgram == "" {
 		return nil, fmt.Errorf("at least one of UDP/TCP/Unixgram listeners must be used")
 	}
-	statsdMapper := &mapper.MetricMapper{MappingsCount: m.MappingsCount}
+	statsdMapper := &mapper.MetricMapper{
+		MappingsCount: m.MappingsCount,
+		Logger:        log,
+	}
```
Comment (rfratto) [inline pkg/integrations/statsd_exporter/statsd_exporter.go:131]: Are there any other fields this needs to have filled out so it doesn't panic? The struct exports quite a bit.

**Review by rfratto — APPROVED** [review]

**Comment by rfratto** [comment]
(I edited your PR description to say `Fixes #1052` so it'll automatically close that issue once this is merged)

File: pkg/integrations/statsd_exporter/statsd_exporter.go (hunk 2 above)
Comment (rfratto) [reply pkg/integrations/statsd_exporter/statsd_exporter.go]: Nope, these are the only fields statsd_exporter fills out.

**Review by mattdurham — APPROVED** [review]
LGTM for the version change in statsd_exporter

**Pull request merged**